distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -1772,11 +1776,14 @@ type Table
        > Example
          Group by the Key column, count the rows
 
-              table.aggregate [Aggregate_Column.Group_By "Key", Aggregate_Column.Count]
+              table.aggregate ["Key"] [Aggregate_Column.Count]
+    @group_by Widget_Helpers.make_column_name_vector_selector
     @columns Widget_Helpers.make_aggregate_column_vector_selector
-    aggregate : Vector Aggregate_Column -> Boolean -> Problem_Behavior -> Table ! No_Output_Columns | Invalid_Aggregate_Column | Invalid_Column_Names | Duplicate_Output_Column_Names | Floating_Point_Equality | Invalid_Aggregation | Unquoted_Delimiter | Additional_Warnings
-    aggregate self columns (error_on_missing_columns=False) (on_problems=Report_Warning) =
-        validated = Aggregate_Column_Helper.prepare_aggregate_columns self.column_naming_helper columns self error_on_missing_columns=error_on_missing_columns
+    aggregate : Vector (Integer | Text | Regex | Aggregate_Column) | Text | Integer | Regex -> Vector Aggregate_Column -> Boolean -> Problem_Behavior -> Table ! No_Output_Columns | Invalid_Aggregate_Column | Invalid_Column_Names | Duplicate_Output_Column_Names | Floating_Point_Equality | Invalid_Aggregation | Unquoted_Delimiter | Additional_Warnings
+    aggregate self group_by columns=[] (error_on_missing_columns=False) (on_problems=Report_Warning) =
```

Comment:
Should the arguments maybe be swapped?

IMHO
```
table.aggregate [Aggregate_Column.Sum "Price"] group_by=["Customer ID"]
```
reads nicely and is actually a bit more similar to SQL where you first do aggregates and later groupings.

**Moreover**, this allows us to _simply skip_ the `group_by` argument altogether when we want to _just get the total_ of the whole table without grouping (which is a legitimate use-case of `Table.aggregate` AFAIK).

*Moreover* (although this argument is the weakest as I assume we may retire the compat mode in future), in 'compatibility mode', currently the only used argument is `group_by` and the fact that other aggregations end up in an argument called `group_by` is a bit weird. However, when we swap the args order - all aggregations would go into `columns` as it used to be which feels less weird.

Reply:
I think `group_by` should be first (same order as in `cross_tab`). I think SQL is wrong but we are stuck with that one.
These columns are strictly included in the output at the beginning as well.

With the graphical design flow, skipping the argument is built in.

Reply:
> These columns are strictly included in the output at the beginning as well.

That is a good argument to the contrary indeed, I wanted to mention it but forgot.

> With the graphical design flow, skipping the argument is built in.

Ah, I did not consider that. I guess in such case the current order of args is indeed alright. 👍